R/string_operations.r

```diff
@@ -212,6 +214,14 @@ col_strsplit <- function(x, split, ...) {
   plain <- strip_style(x)
   splits <- re_table(split, plain, ...)
   chunks <- non_matching(splits, plain, empty = TRUE)
+  # drop any zero length trailing chunks that are created when the split
+  # happens at end of string; this is to replicate 'substr' behavior
+  chunks <- lapply(
+    chunks,
+    function(y)
+      if(nrow(y) > 1L && !tail(y, 1L)[, "length"]) head(y, -1L) else y
```

Comment:
Why the `nrow(y) > 1L` condition? I have 
```r
❯ strsplit("", "a")[[1]]
character(0)
```
so we want to drop a single `""` as well, I think.